File: backend/app/models/schemas/telegram.py
"""
ChatterMate - Telegram Schemas
Copyright (C) 2024 ChatterMate

This program is free software: you can redistribute it and/or modify
it under the terms of the GNU Affero General Public License as
published by the Free Software Foundation, either version 3 of the
License, or (at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU Affero General Public License for more details.

You should have received a copy of the GNU Affero General Public License
along with this program.  If not, see <https://www.gnu.org/licenses/>
"""

from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime


class TelegramConnectRequest(BaseModel):
    """Request schema for connecting a Telegram bot."""
    bot_token: str
    agent_id: str


class TelegramConfigResponse(BaseModel):
    """Response schema for a Telegram bot configuration."""
    id: int
    organization_id: str
    agent_id: str
    bot_username: str
    bot_display_name: Optional[str] = None
    is_active: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class TelegramStatusResponse(BaseModel):
    """Response schema for Telegram connection status."""
    connected: bool
    configs: List[TelegramConfigResponse] = []

File: frontend/src/services/telegram.ts
import api from './api'

// Types
export interface TelegramConfig {
  id: number
  organization_id: string
  agent_id: string
  bot_username: string
  bot_display_name: string | null
  is_active: boolean
  created_at: string
  updated_at: string
}

export interface TelegramStatusResponse {
  connected: boolean
  configs: TelegramConfig[]
}

export async function connectTelegramBot(agentId: string, botToken: string): Promise<TelegramConfig> {
  const response = await api.post('/telegram/connect', {
    bot_token: botToken,
    agent_id: agentId
  })
  return response.data
}

export async function getTelegramStatus(): Promise<TelegramStatusResponse> {
  const response = await api.get('/telegram/status')
  return response.data
}

export async function disconnectTelegramBot(configId: number): Promise<void> {
  await api.delete(`/telegram/${configId}`)
}
